numpy/lib/tests/test__datasource.py

```diff
@@ -26,19 +26,14 @@ def urlopen_stub(url, data=None):
     else:
         raise URLError('Name or service not known')
 
-# setup and teardown
-old_urlopen = None
-
-
-def setup():
-    global old_urlopen
-
-    old_urlopen = urllib_request.urlopen
-    urllib_request.urlopen = urlopen_stub
 
+@pytest.fixture(autouse=True)
```

Comment:
This is a bit risky since patching stdlib functions can break pytest itself--that's why I used the context manager. Here you'll monkeypatch all tests instead of just a few targeted ones.

Reply:
It's true it's in the stdlib, but it didn't seem like pytest would have much use for `urlopen`.

An autouse fixture in a module only applies to the module, so it is no different than the (intended effect of the) previous code. Though I'm not sure the exact scope of `setup`/`teardown`, so it's possible this fixture may run more times than that used to (since I specified no scope.)